### PR TITLE
draft: implementation of lightdash connector.

### PR DESCRIPTION
**Pull request by gauthk6**
### Describe your changes:

Fixes <8166>

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ X] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ X] My PR title is `Fixes <issue-number>: <short explanation>`
- [ X] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/dashboard/lightdash/metadata.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            try:
+                yield from self._yield_lineage_from_query(
+                    chart_details=chart,
+                    db_service_name=db_service_name,
+                    dashboard_name=dashboard_name,
+                ) or []
+
+            except Exception as exc:  # pylint: disable=broad-except
+                logger.debug(traceback.format_exc())
+                logger.error(f"Error creating chart [{chart}]: {exc}")
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/dashboard/lightdash/metadata.py:184]: you can remove this for now and just leave the doc string. We can work on the implementation later.

File: ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    def get_charts_list(self) -> List[LightdashChart]:
+        """
+        Get List of all charts
+        """
+        try:
+            response = self.client.get(
+                f"api/v1/projects/{self.config.projectUUID}/charts"
+            )
+            response_json_results = response["results"]
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py:60]: We should do something like `results = response.get("results")` and then below check if `res` is None. Other we could end up with `Object of type "None" is not subscriptable` -- unless we have a guarantee on the API side to always have a `results` field

File: ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    def get_dashboards_list(self) -> List[LightdashDashboard]:
+        """
+        Get List of all charts
+        """
+
+        try:
+            response = self.client.get(
+                f"api/v1/projects/{self.config.projectUUID}/spaces/{self.config.spaceUUID}"
+            )
+            results = response["results"]
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py:83]: same as above

File: ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            logger.warning(
+                "Failed to fetch the dashboard list for the Lightdash Connector"
+            )
+        return []
+
+    def add_dashboard_lineage(self, dashboards_list) -> None:
+        charts_uuid_list = []
+        for dashboard in dashboards_list:
+            response = self.client.get(f"api/v1/dashboards/{dashboard.uuid}")
+            response_json_results = response["results"]
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py:104]: same as above

File: ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py (hunk 2 above)
Comment (gauthk6) [reply ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py]: done!

File: ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py (hunk 3 above)
Comment (gauthk6) [reply ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py]: done!

File: ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py (hunk 4 above)
Comment (gauthk6) [reply ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py]: done!

**Comment by TeddyCr** [comment]
@gauthk6 once you fix the conflict with `openmetadata-docs/content/v0.13.4/developers/contribute/build-code-and-run-tests/openmetadata-server.md` we can merge the PR 🙂

**Review by TeddyCr — APPROVED** [review]

**Comment by yu-iskw** [comment]
@gauthk6 Thank you for addressing this. I'm interested in how it actually works. What information can we collect from Lightdash? And I can help to enhance APIs to collect more information, if there is anything we want to support more in OpenMetadata.

https://github.com/lightdash/lightdash/issues/5010

**Comment by Prajwal214** [comment]
@gauthk6 The documentation for the Lightdash Dashboard Connector is currently missing. Could you please assist us in adding this connector to the documentation? Your help would be greatly appreciated.

**Pull request merged**